src/repositories/versionable/VersionableRepository.ts

```diff
@@ -19,7 +22,7 @@ export default class VersionableRepository<D extends mongoose.Document, M extend
         return this.model.findOne(query).lean();
     }
 
-    public createUser(data: any, creator): Promise<D> {
+    public async createUser(data: any, creator): Promise<D> {
```

Comment:
Use generalize function name like crate(), instead of createUser(), Since this VersionableRepository can be used with different models, So using one model specific name is not appropriate. Use only generic function names in VersinabeRepositry like create, update,findOne, findAll, count, etc.